File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.ts
function numPairsDivisibleBy60(time: number[]): number {
    let res = 0;
    let c = new Array(60).fill(0);

    for (let t of time) {
        res += c[(60 - t % 60) % 60];
        c[t % 60]++;
    }

    return res;
};

File: 1010-pairs-of-songs-with-total-durations-divisible-by-60/1010-pairs-of-songs-with-total-durations-divisible-by-60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        vector<int> count(60);
        int res = 0;
        for(int t : time) {
            res += count[(600 - t)%60];
            count[t%60] += 1;
        }
        return res;
    }
};